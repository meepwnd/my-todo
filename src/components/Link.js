import React from 'react'
import PropTypes from 'prop-types'

const Link = ({children, active, setVisibilityFilters}) => (
    <button 
        disabled={active}
        onClick={setVisibilityFilters}
    >
        {children}
    </button>
)

Link.propTypes = {
    children: PropTypes.node.isRequired,
    active: PropTypes.bool.isRequired,
    setVisibilityFilters: PropTypes.func.isRequired
}

export default Link;