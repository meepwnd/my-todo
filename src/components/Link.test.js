import React from "react";
import { shallow } from "enzyme";
import Link from "./Link";

describe("link component", () => {
  const setVisibilityFilters = jest.fn();
  const props = {
    active: true,
    children: "test",
    setVisibilityFilters
  };
  const wrapper = shallow(<Link {...props} />);

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call setVisibilityFilters function on button click", () => {
    wrapper.find("button").simulate("click");
    expect(setVisibilityFilters).toHaveBeenCalled();
  });
});
