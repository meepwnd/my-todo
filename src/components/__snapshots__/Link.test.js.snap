// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link component should render correctly 1`] = `
<button
  className="btn btn-filter"
  disabled={true}
  onClick={[MockFunction]}
>
  test
</button>
`;
